### PR TITLE
Split extension code and viewer code

### LINKED ISSUE #10: Split extension code and viewer code

### PR DESCRIPTION
**Pull request by nicolaserny**
fixes #10

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,4 +12,4 @@ settings.json
 out/
 dist/
 .vscode-test
-package-lock.json
```
Comment (mbarekh) [inline .gitignore]: Why did you remove package-lock.json ? It is a generated file

Comment (nicolaserny) [inline .gitignore]: This new feature of npm guarantees that each developer builds the project with the same package versions. So it is a best practice to push this file in our git repository.
[https://stackoverflow.com/questions/44297803/i-am-confused-package-lock-json-role](https://stackoverflow.com/questions/44297803/i-am-confused-package-lock-json-role)
